pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -282,7 +283,13 @@ protected void internalDeleteNamespace(AsyncResponse asyncResponse, boolean auth
                 log.debug("Found topics on namespace {}", namespaceName);
             }
             boolean hasNonSystemTopic = false;
-            for (String topic : topics) {
+            for (String topic : partitionedTopicNames) {
+                if (!pulsar().getBrokerService().isSystemTopic(TopicName.get(topic))) {
+                    hasNonSystemTopic = true;
+                    break;
+                }
+            }
+            for (String topic : topicNames) {
```

Comment:
We don't need go through partitionedTopicNames here.

Reply:
We need this check because users possibly creates and delete topics like this: 

```
1. create-partitioned-topic persistent://my-tenant/my-namespace/my-topic --partitions 2
2. delete persistent://my-tenant/my-namespace/my-topic-partition-0
3. delete persistent://my-tenant/my-namespace/my-topic-partition-1
```

Now, the topic is not actually removed.